### ISSUE TITLE
[SUGGESTION]/[FEEDBACK] - Heat system...

### ISSUE THREAD

**Issue by nokturnihs** [tsunamayo/Skywanderers#1460]
### Short Description of the issue:
1. Same problem as Starmade - Discourages small vessels
2. Discourages using micro-blocks as they do not dissipate heat.
3. Encourages MASSIVE vessels (more hull/armor, more power, more shields, etc)

So here's an alternative - have heat impact power recharge rate or generation overall. If using the 1 to 7 ratio, systems 1-5 don't impact your heat, but if you Max out those systems at 7 you'll be cutting your energy regeneration in half.

That's just an example but balancing heat against full hull blocks is going to push the same kinds of problems Starmade has. When you're talking already of shield/power/thrusters/capacitors being full size blocks you're making mandatory minimum  of 8ish minimum full blocks before we get to any kind of seats, cockpits, etc. That by itself makes fighters much more difficult as weapons will expand that by a lot. 
...

**Comment by nokturnihs** [comment]
One other comment regarding balance systems - people seem to think that there needs to be more complications to the power vs system mechanic - that's not necessarily true. More systems > more power > more blocks > more cost. It doesn't need to be more complex than that. As far as tiers of power are concerned, same equation is just fine, as a fuel or high-tech reactor will have it's own cost above and beyond basic reactors that balances the equation. If for example you wanted to put the top-tier "power of a star" reactor in a fighter, go for it. It's gonna be the most expensive fighter in existence but sure. Don't get shot because it's gonna be expensive to fix if damaged or destroyed. Doesn't need heat, stabilizers or other tertiary systems that add needless complexity that inhibits creativity and defies KISS. That's all.

**Comment by tsunamayo** [comment]
Okay so initially heat was added to prohibit system only ship. It will allow player to have some creative leeway when building their ship, to make some interior ect. Now I am not sure on what Starmade is doing, but it was one critism people had that the heat system was supposed to address. (being forced to make system heavy ship). Also I hope heat is gonna add an interesting dimension to weapons, for example firing a laser will add a bit of heat to your own ship (right now 25% of damage dealt to other ship).

For the small ship - a brick is 1/64 time smaller than a block. If I make brick more powerful than that then I kinda "force" players to use bricks instead of blocks, which is bad for performance. So at the end a small brick ship is always gonna be ridiculously underpowered compared to a medium block ship. This is why I plan to have small brick fighter to fit in a different way into the combat balance. Not sure how yet, but I have several ideas / options. I will focus first on med/large block ship, and see how can small figther could disrupt that a bit. But dont expect to take down 200m ship in a small 10m fighter...

Hope that clarify a bit. At the end I might remove heat entirely if that dont add anything to the game... we will see how it goes!

**Comment by nokturnihs** [comment]
I can understand the concern on system heavy ships - that too was an issue with Starmade (which has a problem of forcing big ships in general, trivializing and discouraging any "small builds" outside of drones).

So maybe heat could be dispersed in some other way that doesn't force any block size (full or small) per se, maybe based on ship mass - system blocks for example or something? Maybe interior "empty space" or something like that? I'm not sure - I see your point but also at the end of the day, you're trying to balance two different play styles - a PvE/RP player isn't going to make system ships while a PvP player is going to go minimal on decor because it'll just get shot up... not sure there IS a way to balance it for both. 

I guess one option would be to make it a system you could toggle on/off in server configs?

**Comment by nokturnihs** [comment]
Just a random side-note regarding weapons - couldn't heat effect energy/system recharge rates? I agree that could be a cool dimension (especially if say lasers hitting your shields or missiles or whatever added heat to your system) - would be a good balance for the PVP side of things... another option I just thought of was if you need to disperse heat using system blocks, could it scale with the size of the ship? For example, a small racer with a lot of engines might only require one or two full blocks to disperse heat while a cruiser would require tons to disperse heat in order to maintain... just throwing wrenches man and hoping it helps. Sorry if I'm frustrating you.

**Comment by Lsutehall** [comment]
Tsuna, for clarification how big do you mean when you say medium and large ships please?

**Comment by tsunamayo** [comment]
@nokturnihs yes that was also one possibity, right now weapon cooldown is linked to energy recharge. But thats something I could explore in the future. No pb I like the discussion ;) !

@Lsutehall so the idea is block base, not brick based. So I think you need at least 20m to have an interesting design I guess, but 50m would be what I call medium.

**Comment by Dwarf-LordPangolin** [comment]
I think I understand what Tsuna's aiming for in terms of ship size vs. heat now; in a nutshell, we're going to have to get used to building much bigger fighters.

A lot of people -- including me -- have been assuming that small ships in Skywanderers were meant to be built on a Star Wars scale, or even smaller -- that is, about this big relative to a person:
![Star-Wars-X-Wing-2015](https://user-images.githubusercontent.com/33041786/66339590-118de480-e909-11e9-9fd4-9c9a14adcd73.jpg)
The X-Wing's fuselage is only about 2 meters high; this is why a lot of us have been building fighters that have block systems, covered with a brick-thick skin, because building on that scale doesn't permit making the hull out of blocks. For a lot of us, something the size of the X-wing -- about 10m x 10m -- is a reasonably-sized fighter, like [this](https://skywarehouse.s3.amazonaws.com/media/covers/20190608003908_1.jpg).

Tsuna, however, seems to have a different ship scale in mind, which is more like the scale of ships in Elite: Dangerous:
![FyUtq3X](https://user-images.githubusercontent.com/33041786/66339598-15216b80-e909-11e9-9f6a-f8b57eb9f328.jpg)
On this scale, the ship is easily large enough to have a block-thick hull, while still having plenty of internal room for systems. As Tsuna has stated, he considers a 20m ship (roughly the dimensions of the E:D Sidewinder above) to be the **minimum** needed for a decent amount of detail, with fighter size being more in line with [this beautiful thing](https://skywarehouse.s3.amazonaws.com/media/covers/20190622120038_1.jpg).

Tsuna's scale, in which fighters primarily use blocks for most functional systems, and bricks are used more sparingly for fine detail or for the systems of _really_ small things like flying cars and speeder bikes, is the correct one for Skywanderers (and not just because it's his game ;) ). The alternative -- making the game so that all-brick fighters would be viable -- would require Tsuna to balance brick systems such that they would be balanced not just for flying cars, speeder bikes, and power loaders, but also with fighter craft. This is impossible: either the speeder bikes and such would be as powerful as small fighter craft, or the fighters would be as weak as motorcycles with machine guns strapped to them.

The reason heat seems too restrictive for small ships is because we've been putting engine systems meant for a fighter on hull sizes that are meant for flying cars; in other words, they overheat for the same reason that a Volvo would overheat if you put the engines and gun from an F-14 on it.

TL;DR: we've been trying to build fighters too small for the kinds of systems we want on them. We'll still be able to make vehicles that are as small as we've been making it, but they won't be fighters. We can't give a 10m ship fighter-grade engines, shields, weapons, etc, without it blowing up from too much heat.

**Comment by nokturnihs** [comment]
That's the big problem with Starmade however - one of the reasons Tsuna's game was so appealing was small ship creation. Starmade put in these kinds of systems to balance PvP (flying system-only bricks) and it ended up killing the game for the most part. Tsuna's gonna have to choose what to balance towards I guess - PvE or PvP - if he chooses the latter it'll make me sad. I understand some of the reasons to discourage small-brick ships but that very option is what attracted me to his game to begin with. Just being honest here and putting in my two cents. @dwarf-lord you're Input is solid though, and well reasoned.

**Comment by nokturnihs** [comment]
So if the heat system was more like 1 system per 1-2 blocks I'd be better with it. Right now every single ship I've made will explode when I enter the game at a guess because I have been building small 10-20 meter ships. They have full blocks but are also jammed with system blocks - lots of engines and power reactors and relatively light on weapons. Doing the math: 1 reactor+1 capacitor+1 shield+1 engine means a minimum of 20 full hull blocks, so bare minimum for a box would be like 4x4x2 for a mostly useless vessel. Doable I guess, but sure limits my creativity if I wanna build small. 

Sorry, know I'm ranting - small ships is THE feature that inspired me to back Skywanderers so to see a PvP balance system in the same vein as Starmade's various broken "balance systems" come in here, kinda frustrating.

**Comment by Dwarf-LordPangolin** [comment]
First, thank you for your kind words, and likewise! And yes, I'm in the same boat with regard to small size and exploding.
![20191007133102_1](https://user-images.githubusercontent.com/33041786/66349460-9800f100-e91e-11e9-8fb7-c81fa195aa19.jpg)

I need to ask a question for clarification: is your main concern that ships will be just plain too big in an objective sense, or that the balance systems Tsuna's been working towards will force builders to chose between ships that fly good, or ships that look good, at that scale? That is, do you just not like the idea of fighters being about 20m long, or are you concerned that 20m fighters built mainly with blocks will look like flying bricks?

**Comment by nokturnihs** [comment]
So it's a little of both but also that system blocks are forcing design choices based around PvP balance. Most of my ships are at least 14m+ long (I might have one or two smaller - or had anyway) - so 20m while huge still in my mind (that's 60ft long - pretty big) - a somewhat functional ship at that size will not have many design choices. One single engine likely won't do all that much, or a single shield block or a single reactor, but each require 5mx1m in full hull blocks, and so that really limits what we can do design wise. Micro-blocks are 1/64th the size (and strength?) As full hull blocks and so the balance in smaller vessels is kinda automatic. You also have each system block taking 1m*1m anyway so if you want to go super small you're not going to have much to work with anyway shield/weapon/power wise so it's kinda balancing it out. Should be faster sure, but a paper doll still won't go toe-to-toe with a corvette or destroyer. I don't have anything against larger ships either but size balances itself. I'd rather see system blocks be balanced via weight or power consumption or something else. And Ultimately, if people want flying system bricks to PvP we should let them. If people want ultra-detailed RP ships, they should get to as well. The incentive of building flying system bricks of doom is they do better in PvP - but PvE is a big thing too and they're balanced completely against each other. One standard on things always breaks the other. Some of the best balanced PvE+PvP games make certain weapons/skills/etc operate VERY differently against players than they do AI mobs.... So as a guy who's pretty much not into (at all) PvP, I don't want to see yet another great building game broken because of PvP balance mechanics. I can list the number of games that were great in EA or fresh releases that were absolutely ruined due to PvP balances. I think the best option on games like this one is to build for PvE And then adds server config options for PvP for those who want them. But I'm not sure where Tsuna is planning on going with Skywanderers - if it's a live service thing that won't work. If it's a host/client thing then sure. Either way, the heat mechanic in it's current state is forcing build/design choices and that kind of counteracts the point of a game where creativity is encouraged.

**Comment by nokturnihs** [comment]
Sorry. I'm a verbal MoFo ... I ramble a little...

**Comment by nokturnihs** [comment]
Sorry, have a bit more to point out - when your car overheats you can screw up your engine but it'll never explode. Your phone can explode under very specific conditions (the battery) but most well-functioning computers will simple burn out or shutdown if they overheat. Heat causing a ship to explode will also make building a serious pain in the behind in some cases. As I said earlier, heat as a mechanic would do better for disabling a ships power production than destroying it outright. I'd rather see disabling/destroying a ship via destroying enough reactors or system blocks than through heat or hull destruction. Just me though.

**Comment by tsunamayo** [comment]
Okay sorry I havent read all the thing yet but it seems there is a misunderstanding, just to make clear I dont want people to discourage to build small fighter, I love them also ! @nokturnihs @Dwarf-LordPangolin 
My plan is:
* Start balancing the game for medium / large block based ship. This is where the bulk of the game progression will take place.
* Only when things settle a bit more I will balance and add small fighters into the mix. They are just too specific to be simply considered under the same umbrella as other ship. If I where to simply scale them proportionally to their size like what you are asking they would be super weak and unusable gameplay wise (a brick is 1/64 of a block). Also I just cant as my damage / hull model is voxel based.

Thats it. That is just a temporary thing, I need to start somewhere, I simply start where it makes the most sense. So if you want to help balancing the game you will have to build bigger ship for now, otherwise that s fine you can still make good looking fighter ;)

**Comment by nokturnihs** [comment]
So can you instead roll heat to overloading/disabling ship systems? I've got to admit I found it curious why ship destruction is based on hull/heat not system functionality. I'd guess it's system heavy to try and process determine ship destruction by checking system block integrity but would be interested in why you chose hull blocks and heat... it's very different from what I expected (although I do like that you don't need to destroy the whole ship).

**Comment by tsunamayo** [comment]
Not sure to understand, what do you expected to see regarding heat and system? Namely:
1) What is producing heat,
2) What is dissipating heat, 
3) In what regard having too much heat impact the ship?
Cheers

**Comment by tsunamayo** [comment]
Okay so for the time being I made a branch with a build roll back for people who want to continue working on brick based ship. #1500 
Thanks

**Comment by tsunamayo** [comment]
Okay  so now I have added a surface heat bonus to all ship.
As surface raise with square whereas ship power raise with cube, it will be impactfull for small ship and less for large ship, which is what we want.
Update is already live!
Cheers
